apps/src/clientApi.js

```diff
@@ -171,20 +183,28 @@ class AssetsApi extends CollectionsApi {
       },
       error);
   }
+
+  /*
+   * Create or update an asset
+   * @param filename {String} filename to be created or updated
+   * @param data {Blob} asset data
+   * @param success {Function} callback when successful (includes xhr parameter)
+   * @param error {Function} callback when failed
+   */
+  putAsset(filename, data, success, error) {
+    this._withBeforeFirstWriteHook(() => {
+      ajaxInternal('PUT', this.basePath(filename), xhr => {
+          if (success) {
+            success(xhr);
+          }
+        },
+        error,
+        data);
+    });
+  }
 }
 
 class FilesApi extends CollectionsApi {
-  _withBeforeFirstWriteHook(fn) {
```

Comment:
Feels like `registerBeforeFirstWriteHook` should also move up into CollectionsApi if this helper is moving there.